### PR TITLE
Add prettier

### LINKED ISSUE #172: Add prettier
It would be nice to have consistent formatting that is linted. @domoritz thoughts on adding prettier via eslint to do this?

### PR DESCRIPTION
**Pull request by AndrewLeedham**
I tried to use a prettier config that matched the current codebase as closely as possible. Hence there is not really any changes to the current files, it will just enforce the current style going forward.

I split the change to the `.eslintrc.js` and the formatting into different commits, so it is easier to review.

fixes #172

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -14,6 +14,6 @@ indent_size = 4
 indent_style = space
```
Comment (domoritz) [inline .editorconfig]: should we just get rid of the editorconfig now?

**Comment by AndrewLeedham** [comment]
> I'm impressed by how little changes. I just updated some deps. Can you merge master?

Did seem a bit good to be true. I don't think prettier is working properly.

**Comment by domoritz** [comment]
> Did seem a bit good to be true. I don't think prettier is working properly.

That's what I'm thinking as well ;-)

I remember trying prettier a while ago and it caused a lot of changes I didn't want. However, I also didn't customize the settings and it was a long time ago.

**Comment by AndrewLeedham** [comment]
Fixed it, I forgot to add the `"prettier/prettier"` rule when I split the implementation and formatting in to separate commits 🤦‍♂

File: .eslintrc.js [hunk 2]
```diff
@@ -44,5 +44,6 @@ module.exports = {
         "no-shadow": "error",
         "no-useless-concat": "error",
         "no-undef": "off",
+        "prettier/prettier": "error",
```
Comment (domoritz) [inline .eslintrc.js]: Can we make this a `"prettier/prettier": "warn"` and move it as the first rule?

**Comment by domoritz** [comment]
I guess this will make https://github.com/vega/ts-json-schema-generator/issues/63 much more difficult since a lot of code will change. But then I'm also not sure what we are still missing.

File: .eslintrc.js (hunk 2 above)
Comment (AndrewLeedham) [reply .eslintrc.js]: What is the reasoning for `warn`? `error` will enforce style during CI, so all new code is guaranteed to be consistent.

File: .editorconfig (hunk 1 above)
Comment (AndrewLeedham) [reply .editorconfig]: I think it makes sense to include both, so people using editors without a prettier plugin or vscode will still have some basic styling setup.

**Comment by domoritz** [comment]
It's probably best to look through individual commits anyway. I see the value of prettier so let's go ahead with this.

File: .eslintrc.js (hunk 2 above)
Comment (domoritz) [reply .eslintrc.js]: Hmm, I see. I just don't know how much I will get annoyed by red squiggly lines while coding. Let's leave it for now and see how I feel later.

Comment (AndrewLeedham) [reply .eslintrc.js]: Personally I use the [prettier plugin](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) for vscode as well as format on save options, so as soon as I save the squiggly lines go away 😃

**Pull request merged**